### PR TITLE
refactoring the language research

### PR DESCRIPTION
**Pull request by Maxstgt**
In this PR the language research part got refactored. Now it is like on the movie search page. 
Also, I did some improvements to Victors datepicker component.

### CODE REVIEW

File: libs/ui/src/lib/datepicker/datepicker.component.ts [hunk 1]
```diff
@@ -98,7 +90,7 @@ export class DatepickerRangeComponent {
     snackBarRef.onAction().subscribe(() => this.initDatepicker());
   }
 
-  getErrorMessage() {
+  get getErrorMessage() {
```
Comment (pldespaigne) [inline libs/ui/src/lib/datepicker/datepicker.component.ts]: `get get` is weird

File: apps/catalog/marketplace/marketplace/src/app/distribution-right/create/create.component.ts [hunk 2]
```diff
@@ -185,76 +140,34 @@ export class DistributionRightCreateComponent implements OnInit, OnDestroy {
...
     this.choosenDateRange.to = dates.to;
     this.choosenDateRange.from = dates.from;
   }
 
   public selectedTerritory(territory: MatAutocompleteSelectedEvent) {
-    if (!this.selectedTerritories.includes(territory.option.viewValue)) {
-      this.selectedTerritories.push(territory.option.value);
-    }
-    this.form.addTerritory(territory.option.viewValue as TerritoriesSlug);
+    this.form.addTerritory(getCodeIfExists('TERRITORIES', territory.option.viewValue));
```
[the remark this answers is not in the document]
Comment (Maxstgt) [reply apps/catalog/marketplace/marketplace/src/app/distribution-right/create/create.component.ts]: Then it returns null and the form will throw an error

File: apps/catalog/marketplace/marketplace/src/app/distribution-right/create/create.component.ts [hunk 3]
```diff
@@ -317,8 +229,12 @@ export class DistributionRightCreateComponent implements OnInit, OnDestroy {
...
 
           if (mediasNotInSalesAgentScope.length || territoriesNotInSalesAgentScope.length) {
-            // @todo #980 let customer make a request to sales agent ( to see if he can adapt and find a solution)
-            console.log('Some territories or medias are not in sales agent scope', territoriesNotInSalesAgentScope, mediasNotInSalesAgentScope);
+            // @todo #1036 let customer make a request to sales agent ( to see if he can adapt and find a solution)
+            console.log(
+              'Some territories or medias are not in sales agent scope',
+              territoriesNotInSalesAgentScope,
+              mediasNotInSalesAgentScope
+            );
```
[the remark this answers is not in the document]
Comment (Maxstgt) [reply apps/catalog/marketplace/marketplace/src/app/distribution-right/create/create.component.ts]: No this is our error handling for now, until the new UI/UX is done.

File: libs/utils/src/lib/form/validators/validators.ts [hunk 4]
```diff
@@ -114,7 +114,7 @@ export function numberRangeValidator(from: string, to: string): ValidatorFn {
   return (group: FormGroup): ValidationErrors => {
     const control1 = group.controls[from];
     const control2 = group.controls[to];
```
[the remark this answers is not in the document]
Comment (Maxstgt) [reply libs/utils/src/lib/form/validators/validators.ts]: you will get shadow naming. parameters and variable names. I will choose a different name

File: apps/catalog/marketplace/marketplace/src/app/movie/search/search.form.ts [hunk 5]
```diff
@@ -76,10 +78,10 @@ function createCatalogSearch(search: Partial<CatalogSearch>): CatalogSearch {
 /* CREATE CONTROL */
 /* -------------- */
 
-function createLanguageControl(language: MovieLanguage) {
+export function createLanguageControl(language: MovieLanguageSpecification, disableDubbed?: boolean) {
   return new FormGroup({
     original: new FormControl(language.original),
-    dubbed: new FormControl(language.dubbed),
+    dubbed: new FormControl({ value: language.dubbed, disabled: disableDubbed }),
```
[the remark this answers is not in the document]
Comment (Maxstgt) [reply apps/catalog/marketplace/marketplace/src/app/movie/search/search.form.ts]: yes and undefined will be then false, which is expected

**Pull request merged**